src/netaddress.h

```diff
@@ -477,6 +480,22 @@ class CNetAddr
         }
 };
 
+class CNetAddrHash
+{
+public:
+    size_t operator()(const CNetAddr& a) const noexcept
+    {
+        CSipHasher hasher(m_salt_k0, m_salt_k1);
+        hasher.Write(reinterpret_cast<const uint8_t*>(&a.m_net), sizeof(a.m_net));
```

Comment:
Micronit: this reinterpret cast is a bit ugly. You could avoid it by just doing `hasher.Write(a.m_net);`; it'll interpret `a.m_net` as a uint64_t and use the optimized integer hasher.

Reply:
Done. I hope the implicit conversion from `int` to `uint64_t` will not upset some compiler.

Reply:
The ugliness could maybe be considered a virtue; it's explicit that this is an *"on your head be it!"* type of cast and highlights that the code could be improved. https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast

Reply:
Here we can do without a cast since there is `Write()` method that takes an integer argument (modulo `int` -> `uint64` conversion which is ok because we never pass negative values).